src/Command/ListDatabases.php

```diff
@@ -0,0 +1,154 @@
+     *
+     * @param array $options Command options
+     * @throws InvalidArgumentException for parameter/option parsing errors
+     */
+    public function __construct(array $options = [])
+    {
+        if (isset($options['authorizedDatabases']) && ! is_bool($options['authorizedDatabases'])) {
+            throw InvalidArgumentException::invalidType('"authorizedDatabases" option', $options['authorizedDatabases'], 'boolean');
+        }
+
+        if (isset($options['filter']) && ! is_array($options['filter']) && ! is_object($options['filter'])) {
+            throw InvalidArgumentException::invalidType('"filter" option', $options['filter'], ['array', 'object']);
+        }
+
+        if (isset($options['maxTimeMS']) && ! is_integer($options['maxTimeMS'])) {
+            throw InvalidArgumentException::invalidType('"maxTimeMS" option', $options['maxTimeMS'], 'integer');
+        }
+
+        if (isset($options['nameOnly']) && ! is_bool($options['nameOnly'])) {
+            throw InvalidArgumentException::invalidType('"nameOnly" option', $options['nameOnly'], 'boolean');
+        }
+
+        if (isset($options['session']) && ! $options['session'] instanceof Session) {
+            throw InvalidArgumentException::invalidType('"session" option', $options['session'], Session::class);
+        }
+
+        $this->options = $options;
+    }
+
+    /**
+     * Execute the operation.
+     *
+     * @see Executable::execute()
+     * @param Server $server
+     * @return array An array of database names or an array of database info structures, depending on the nameOnly option
+     * @throws UnexpectedValueException if the command response was malformed
+     * @throws DriverRuntimeException for other driver errors (e.g. connection errors)
+     */
+    public function execute(Server $server)
+    {
+        $cmd = ['listDatabases' => 1];
+
+        if (isset($this->options['authorizedDatabases'])) {
+            $cmd['authorizedDatabases'] = $this->options['authorizedDatabases'];
+        }
+
+        if (! empty($this->options['filter'])) {
+            $cmd['filter'] = (object) $this->options['filter'];
+        }
+
+        if (isset($this->options['maxTimeMS'])) {
+            $cmd['maxTimeMS'] = $this->options['maxTimeMS'];
+        }
+
+        if (isset($this->options['nameOnly'])) {
+            $cmd['nameOnly'] = $this->options['nameOnly'];
+        }
+
+        $cursor = $server->executeReadCommand('admin', new Command($cmd), $this->createOptions());
+        $cursor->setTypeMap(['root' => 'array', 'document' => 'array']);
+        $result = current($cursor->toArray());
+
+        if (! isset($result['databases']) || ! is_array($result['databases'])) {
+            throw new UnexpectedValueException('listDatabases command did not return a "databases" array');
+        }
+
+        // Using array_column here emulates nameOnly behavior for servers that don't support it and ensures a consistent return type
```

Comment:
This is what I'd suggest moving to ListDatabaseNames, which currently just yields from this result directly. I suppose doing `yield from array_column($result['databases'], 'name')` would be cheaper than using the CallbackIterator, especially since the result is already an array (unlike a potentially large listCollections response, where iterators may conserve memory).

Reply:
`array_column` won't work here since the `ListCollections` command returns a CachingIterator over the result cursor, which is why I added the iterator on top of that.

Reply:
We're talking about ListDatabases here, though, which always returns an array. Am I misunderstanding?

Reply:
Apologies, I got confused. `ListDatabaseNames` uses `array_column` since we're already dealing with an array.